pkg/ssh/ssh.go

```diff
@@ -233,3 +243,43 @@ func extractOptionFromFlag(flag string) (string, string, error) {
 	}
 	return oParts[0], oParts[1], nil
 }
+
+// openssh must be running 9.2 or above
+// checks version in format OpenSSH_{MAJOR}.{MINOR}
+func validateSSHVersion(ctx context.Context, sshCmd string) error {
+	version, err := getSSHVersion(ctx, sshCmd)
+	if err != nil {
+		return err
+	}
+	return RequireSSHVersionAbove9_2(version)
+}
+
+func getSSHVersion(ctx context.Context, sshCmd string) (string, error) {
+	out, err := exec.CommandContext(ctx, sshCmd, "-V").CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("failed to run ssh -V command: %w", err)
+	}
+	return string(out), nil
+}
+
+func RequireSSHVersionAbove9_2(version string) error {
```

Comment:
If you want to, this seems like a good candidate for property based testing (generating bunch of random inputs in a specific format)

I usually find bugs doing this

Reply:
Cool, I was reading about it. In this case, the edge cases are very strict around 9.2 which is fairly simple to test anyway.